### PR TITLE
MATLAB script-based MASSIF API #63

### PR DESCRIPTION
**Pull request by lunkpeter**
See details in #63

### CODE REVIEW

**Comment by abelhegedus** [comment]
Jenkins, please add to whitelist

File: plugins/hu.bme.mit.massif.simulink.cli/src/hu/bme/mit/massif/simulink/cli/CLIEMFCreator.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2010-2013, Embraer S.A., Budapest University of Technology and Economics
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.cli/src/hu/bme/mit/massif/simulink/cli/CLIEMFCreator.java]: Please correct copyright: use 2010-2017 as date and your own employer as holder.

File: plugins/hu.bme.mit.massif.simulink.cli/src/hu/bme/mit/massif/simulink/cli/CLIEMFCreator.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * All rights reserved. This program and the accompanying materials 
+ * are made available under the terms of the Eclipse Public License v1.0 
+ * which accompanies this distribution, and is available at 
+ * http://www.eclipse.org/legal/epl-v10.html 
+ *
+ * Contributors: 
+ *     Marton Bur, Abel Hegedus, Akos Horvath - initial API and implementation
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.cli/src/hu/bme/mit/massif/simulink/cli/CLIEMFCreator.java]: I'm pretty sure we didn't write this code 😃

File: plugins/hu.bme.mit.massif.simulink.cli/src/hu/bme/mit/massif/simulink/cli/CLIMatlabCreator.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2010-2013, Embraer S.A., Budapest University of Technology and Economics
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.cli/src/hu/bme/mit/massif/simulink/cli/CLIMatlabCreator.java]: Fix copyright

File: plugins/hu.bme.mit.massif.simulink.cli/src/hu/bme/mit/massif/simulink/cli/CLIMatlabCreator.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * All rights reserved. This program and the accompanying materials 
+ * are made available under the terms of the Eclipse Public License v1.0 
+ * which accompanies this distribution, and is available at 
+ * http://www.eclipse.org/legal/epl-v10.html 
+ *
+ * Contributors: 
+ *     Marton Bur, Abel Hegedus, Akos Horvath - initial API and implementation
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.cli/src/hu/bme/mit/massif/simulink/cli/CLIMatlabCreator.java]: Fix author

File: plugins/hu.bme.mit.massif.simulink.cli/META-INF/MANIFEST.MF [hunk 5]
```diff
@@ -0,0 +1,12 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: Massif command line interface
+Bundle-SymbolicName: hu.bme.mit.massif.simulink.cli
+Bundle-Version: 0.6.0.qualifier
+Bundle-RequiredExecutionEnvironment: JavaSE-1.8
+Bundle-Vendor: Budapest University of Technology and Economics
+Require-Bundle: hu.bme.mit.massif.communication,
+ hu.bme.mit.massif.simulink,
+ org.eclipse.emf.ecore.xmi,
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.cli/META-INF/MANIFEST.MF]: Is this requirement needed through some transitive dependency?

**Comment by lunkpeter** [comment]
Jenkins, retest this please

**Comment by lunkpeter** [comment]
Jenkins, retest this please

**Comment by lunkpeter** [comment]
Jenkins, retest this please

**Comment by lunkpeter** [comment]
Jenkins retest this please

**Comment by lunkpeter** [comment]
I have reached a point in the CLI development, where the results could be merged into master.  Please feel free to review the code.

P.S.: The build should succeed once Nexus updates.

**Comment by lunkpeter** [comment]
TODO: version number bump

File: plugins/hu.bme.mit.massif.simulink.cli/scripts/CLI_CreateMATLABModel.m [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+% which accompanies this distribution, and is available at 
+% http://www.eclipse.org/legal/epl-v10.html 
+%
+% Contributors: 
+%     Peter Lunk- initial API and implementation 
+%##############################################################################
+ javaaddpath('hu.bme.mit.massif.simulink.cli-package.jar');
+ 
+ modelImporter = hu.bme.mit.massif.simulink.cli.CLIMatlabCreator;
+ modelImporter.createMatlabModel('EMF_MODEL_NAME', 'EMF_MODEL_LOCATION');
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.cli/scripts/CLI_CreateMATLABModel.m]: Please add a comment that this is an example script for using the CLIMatlabCreator and the placeholders should be replaced with the actual values. Also describe the expected format for the location.

File: plugins/hu.bme.mit.massif.simulink.cli/scripts/CLI_CreateMassifModel.m [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+% which accompanies this distribution, and is available at 
+% http://www.eclipse.org/legal/epl-v10.html 
+%
+% Contributors: 
+%     Peter Lunk- initial API and implementation 
+%##############################################################################
+javaaddpath('hu.bme.mit.massif.simulink.cli-package.jar');
+
+modelExporter = hu.bme.mit.massif.simulink.cli.CLIEMFCreator;
+modelExporter.createSimulinkModel('MODEL_NAME', 'TARGET_LOCATION');
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.cli/scripts/CLI_CreateMassifModel.m]: Please add a comment that this is an example script for using the CLIMatlabCreator and the placeholders should be replaced with the actual values. Also describe the expected format for the location.

File: plugins/hu.bme.mit.massif.simulink.cli/src/hu/bme/mit/massif/simulink/cli/CLIMatlabCreator.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+        Thread thread = new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+                try {
+                    exporter.export(loadedModel, commandFactory);
+
+                    String fqn = loadedModel.getSimulinkRef().getFQN();
+                    exporter.saveSimulinkModel(fqn, "slx");
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.cli/src/hu/bme/mit/massif/simulink/cli/CLIMatlabCreator.java]: I think hardcoding the SLX format here is not a good idea.

**Comment by ujhelyiz** [comment]
Note that not the entire change can be added to the master branch as well - the 'fat jar' generator required query-based features runtime available from the maven repository that did not happen before VIATRA 2.0 (see https://bugs.eclipse.org/bugs/show_bug.cgi?id=527868 for details).

File: plugins/br.com.embraer.massif.commandevaluation.commands/.project [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.sonarlint.eclipse.core.sonarlintBuilder</name>
```
Comment (ujhelyiz) [inline plugins/br.com.embraer.massif.commandevaluation.commands/.project]: Do we use SonarLint for Massif? In my experience, the use of SonarLint slows down compiling very noticeably, so if we don't plan to set it up, we might leave it out from here.

File: plugins/br.com.embraer.massif.commandevaluation.commands/.settings/org.sonar.ide.eclipse.core.prefs [hunk 10]
```diff
@@ -0,0 +1,7 @@
+analyseLocally=false
```
Comment (ujhelyiz) [inline plugins/br.com.embraer.massif.commandevaluation.commands/.settings/org.sonar.ide.eclipse.core.prefs]: Again, do we need sonar configuration. Especially if it is incorrect, e.g. see serverUrl.

File: plugins/hu.bme.mit.massif.communication.localscript/.project [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.pde.SchemaBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.sonarlint.eclipse.core.sonarlintBuilder</name>
```
Comment (ujhelyiz) [inline plugins/hu.bme.mit.massif.communication.localscript/.project]: SonarLint again

File: plugins/hu.bme.mit.massif.communication.localscript/.settings/org.sonar.ide.eclipse.core.prefs [hunk 12]
```diff
@@ -0,0 +1,7 @@
+analyseLocally=false
```
Comment (ujhelyiz) [inline plugins/hu.bme.mit.massif.communication.localscript/.settings/org.sonar.ide.eclipse.core.prefs]: SonarLint again.

File: releng/hu.bme.mit.massif.simulink.cli-package/.project [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.pde.FeatureBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.sonarlint.eclipse.core.sonarlintBuilder</name>
```
Comment (ujhelyiz) [inline releng/hu.bme.mit.massif.simulink.cli-package/.project]: SonarLint again.

File: plugins/br.com.embraer.massif.commandevaluation.commands/.project (hunk 9 above)
Comment (abelhegedus) [reply plugins/br.com.embraer.massif.commandevaluation.commands/.project]: I suppose Péter only added it to keep it consistent with all other places. I agree that we can remove it from the project configurations. However, it doesn't have to be done as part of the PR.

**Comment by lunkpeter** [comment]
I think we should handle these points in a separate issue

**Pull request merged**